src/main/native/pairhmm/IntelPairHmm.cc

```diff
@@ -64,8 +70,21 @@ JNIEXPORT void JNICALL Java_com_intel_gkl_pairhmm_IntelPairHmm_initNative
   _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
 
   // set function pointers
-  g_compute_full_prob_float = compute_full_prob_avxs;
-  g_compute_full_prob_double = compute_full_prob_avxd;
+  if(is_avx512_supported())
+  {
+#ifndef __APPLE__
+    DBG("Using CPU-supported AVX-512 instructions");
+    g_compute_full_prob_float = compute_fp_avx512s;
+    g_compute_full_prob_double = compute_fp_avx512d;
+#else
+    assert(false);
```

Comment:
What does this line do? Throws an assertion when running on OSX and AVX-512 is supported?
Maybe this will never happen with the current OSX compiler options.

Reply:
On OS X, the function `is_avx512_supported()` returns false, so this line should never be reached. (All this Mac-vs-non-Mac logic will be gone once we support AVX-512 on OS X.)